### PR TITLE
User login

### PR DESCRIPTION
**Pull request by eliascodes**
Related to #170, #167 

Update to the login route as outlined in #170 

### Changes
* `Roles.userHas` supports list of roles
* Rename `/users/login/admin` to `users/login`, update logic to support all user types
* Update handler logic to support returning token in payload instead of as a cookie

### CODE REVIEW

File: src/auth/roles/index.ts [hunk 1]
```diff
@@ -85,15 +85,18 @@ const Roles: RolesInterface = {
   },
 
   userHas: async (client, { role, userId, organisationId }) => {
+    const sub = Array.isArray(role)
+      ? client('access_role') .select('access_role_id') .whereIn('access_role_name', role)
```
Comment (astroash) [inline src/auth/roles/index.ts]: Why are you using `whereIn`, are you expecting `role` to be an array?

Comment (eliascodes) [reply src/auth/roles/index.ts]: Yes, hence `Array.isArray`.

Comment (astroash) [reply src/auth/roles/index.ts]: Can it be changed to `roles`?

File: src/api/v1/users/__tests__/login.test.integration.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
+import * as Hapi from 'hapi';
+import * as Knex from 'knex';
+import * as JWT from 'jsonwebtoken';
+import { init } from '../../../../server';
+import { getConfig } from '../../../../../config';
+import { getCookie } from '../../../../utils';
+import { getTrx } from '../../../../../tests/utils/database';
+
+
+describe('POST /users/login/admin', () => {
```
Comment (astroash) [inline src/api/v1/users/__tests__/login.test.integration.ts]: 🔥 'POST /users/login/**admin**'

File: src/auth/roles/index.ts (hunk 1 above)
Comment (eliascodes) [reply src/auth/roles/index.ts]: It's not always an array, hence the ternary.

Comment (astroash) [reply src/auth/roles/index.ts]: Thought it read better as plural, but feel free to leave.

**Pull request merged**